sw/device/tests/clkmgr_external_clk_src_for_sw_impl.c

```diff
@@ -63,14 +57,30 @@ void execute_clkmgr_external_clk_src_for_sw_test(bool fast_ext_clk) {
       dif_clkmgr_external_clock_set_enabled(&clkmgr,
                                             /*is_low_speed=*/!fast_ext_clk));
 
-  // Wait a few AON cycles for glitches from the transition to external
-  // clock to settle.
+  // Wait for the external clock to become active.
   IBEX_SPIN_FOR(did_extclk_settle(&clkmgr), kSettleDelayMicros);
+  LOG_INFO("External clock enabled");
 
   clkmgr_testutils_enable_clock_counts_with_expected_thresholds(
       &clkmgr, /*jitter_enabled=*/false, /*external_clk=*/true,
       /*low_speed=*/!fast_ext_clk);
   busy_spin_micros(delay_micros);
   CHECK(clkmgr_testutils_check_measurement_counts(&clkmgr));
   clkmgr_testutils_disable_clock_counts(&clkmgr);
+
+  // Disable the external clock and check the frequencies are as expected
+  // when driven by the AST oscillators.
+  LOG_INFO("Disabling external clock");
+  CHECK_DIF_OK(dif_clkmgr_external_clock_set_disabled(&clkmgr));
+
+  // Wait for the external clock to become inactive.
+  IBEX_SPIN_FOR(!did_extclk_settle(&clkmgr), kSettleDelayMicros);
+  LOG_INFO("External clock dsabled");
+
+  clkmgr_testutils_enable_clock_counts_with_expected_thresholds(
```

Comment:
The `clkmgr_testutils` has been refactored to return `status_t`, so if you rebase you should get an error here for not checking the return.
```suggestion
  CHECK_STATUS_OK(clkmgr_testutils_enable_clock_counts_with_expected_thresholds(
```

Reply:
Done